CmisSync.Lib/Sync/SyncTriplet/SyncTripletFactory.cs

```diff
@@ -0,0 +1,264 @@
+            // Create local storage 
+            res.LocalStorage = (null == res.DBStorage.DBLocalPath) ? null : new LocalStorageItem (cmisSyncFolder.LocalPath, res.DBStorage.DBLocalPath);
+
+            return res;
+        }
+
+
+        public static SyncTriplet CreateSFGFromRemoteFolder (
+            IFolder remoteFolder,
+            CmisSyncFolder.CmisSyncFolder cmisSyncFolder)
+        {
+
+            SyncTriplet res = new SyncTriplet (IsFolder);
+
+            String remoteRoot = cmisSyncFolder.RemotePath;
+            String remoteFull = remoteFolder.Path;
+            String remoteRelative = remoteFull.Substring (remoteRoot.Length).TrimStart (CmisUtils.CMIS_FILE_SEPARATOR);
+            res.RemoteStorage = new RemoteStorageItem (remoteRoot, remoteRelative, remoteFolder);
+
+            res.DBStorage = new DBStorageItem (cmisSyncFolder.Database, res.RemoteStorage.RelativePath, IsFolder, RemoteToLocal);
+
+            res.LocalStorage = null;
+
+            res.Name = remoteRelative + Path.DirectorySeparatorChar;
+            return res;
+        }
+
+
+        // Create Full Synctriplet, useful when remote has high prioirty, eg: changelog
+        public static SyncTriplet CreateFromRemoteFolder(
+            IFolder remoteFolder,
+            CmisSyncFolder.CmisSyncFolder cmisSyncFolder)
+        {
+
+            SyncTriplet res = CreateSFGFromRemoteFolder (remoteFolder, cmisSyncFolder);
+
+            res.LocalStorage = (null == res.DBStorage.DBLocalPath) ? null : new LocalStorageItem (cmisSyncFolder.LocalPath, res.DBStorage.DBLocalPath);
+
+            return res;
+        }
+
+
+        // Create SFG of synctriplet from local:
+        // LS, DB, ??
+        public static SyncTriplet CreateSFGFromLocalDocument(
+            String localFullPath,
+            CmisSyncFolder.CmisSyncFolder cmisSyncFolder)
+        {
+            SyncTriplet res = new SyncTriplet (IsDocument);
+
+            String localRoot = cmisSyncFolder.LocalPath;
+            String localRelative = localFullPath.Substring (localRoot.Length).TrimStart (CmisUtils.CMIS_FILE_SEPARATOR);
+            res.LocalStorage = new LocalStorageItem (localRoot, localRelative);
+
+            res.DBStorage = new DBStorageItem (cmisSyncFolder.Database, localRelative, IsDocument, LocalToRemote);
+
+            res.Name = localRelative;
+            return res;
+        }
+
+        // Create Full synctriplet
+        // This method is almost useless.
+        public static SyncTriplet CreateFromLocalDocument(
+            String localFullPath,
+            CmisSyncFolder.CmisSyncFolder cmisSyncFolder)
+        {
+            SyncTriplet res = CreateSFGFromLocalDocument(localFullPath, cmisSyncFolder);
+            res.RemoteStorage = new RemoteStorageItem (cmisSyncFolder.RemotePath, res.DBStorage.DBRemotePath, null);
+            return res;
+        }
+ 
+        public static SyncTriplet CreateSFGFromLocalFolder(
```

Comment:
Expanding SFG (or at least explaining it) would make the code more understandable to readers.